### PR TITLE
Schema restore

### PR DESCRIPTION
**Pull request by vargup**
As a part of restore request, added a block to first restore the schema before restoring the snapshot.

### CODE REVIEW

**Comment by verma7** [comment]
@mesosphere-ci retest this please

**Comment by vargup** [comment]
can someone please review and merge the changes.

File: cassandra-executor/src/main/java/com/mesosphere/dcos/cassandra/executor/backup/S3StorageDriver.java [hunk 1]
```diff
@@ -243,7 +236,7 @@ public void download(BackupRestoreContext ctx) throws Exception {
         final AmazonS3Client amazonS3Client = getAmazonS3Client(ctx);
 
         try {
-            if (ctx.getRestoreType().equals("new")) {
+            if (Objects.equals(ctx.getRestoreType(), new String("new"))) {
```
Comment (nickbp) [inline cassandra-executor/src/main/java/com/mesosphere/dcos/cassandra/executor/backup/S3StorageDriver.java]: `"new".equals(ctx.getRestoreType())` ?

File: cassandra-executor/src/main/java/com/mesosphere/dcos/cassandra/executor/tasks/RestoreSnapshot.java [hunk 2]
```diff
@@ -79,7 +81,7 @@ public void run() {
             sendStatus(driver, Protos.TaskState.TASK_RUNNING,
                     "Started restoring snapshot");
 
-            if (context.getRestoreType().equals("new")) {
+            if (Objects.equals(context.getRestoreType(), new String("new"))) {
```
Comment (nickbp) [inline cassandra-executor/src/main/java/com/mesosphere/dcos/cassandra/executor/tasks/RestoreSnapshot.java]: `"new".equals(context.getRestoreType())`?

File: cassandra-executor/src/main/java/com/mesosphere/dcos/cassandra/executor/tasks/RestoreSnapshot.java [hunk 3]
```diff
@@ -125,21 +127,22 @@ public void run() {
...
                                 command);
+                        processBuilder.redirectErrorStream(true);
                         Process process = processBuilder.start();
-                        int exitCode = process.waitFor();
 
-                        String stdout = streamToString(process.getInputStream());
-                        String stderr = streamToString(process.getErrorStream());
+                        BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+                        String line;
+                        while ((line = reader.readLine()) != null)
```
Comment (nickbp) [inline cassandra-executor/src/main/java/com/mesosphere/dcos/cassandra/executor/tasks/RestoreSnapshot.java]: Please add braces to this single loop:

```
while ((line = reader.readLine()) != null) {
    LOGGER.info(line);
}
```

File: cassandra-executor/src/main/java/com/mesosphere/dcos/cassandra/executor/tasks/RestoreSnapshot.java [hunk 4]
```diff
@@ -125,21 +127,22 @@ public void run() {
 
                         final ProcessBuilder processBuilder = new ProcessBuilder(
                                 command);
+                        processBuilder.redirectErrorStream(true);
                         Process process = processBuilder.start();
-                        int exitCode = process.waitFor();
 
-                        String stdout = streamToString(process.getInputStream());
-                        String stderr = streamToString(process.getErrorStream());
+                        BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
```
Comment (nickbp) [inline cassandra-executor/src/main/java/com/mesosphere/dcos/cassandra/executor/tasks/RestoreSnapshot.java]: Looks like `streamToString()` is no longer used and can be removed?

File: cassandra-executor/src/main/java/com/mesosphere/dcos/cassandra/executor/backup/S3StorageDriver.java (hunk 1 above)
Comment (vargup) [inline cassandra-executor/src/main/java/com/mesosphere/dcos/cassandra/executor/backup/S3StorageDriver.java]: in previous PR, @mohitsoni gave me feedback to do object.equals.

**Comment by nickbp** [comment]
retest this please

**Comment by nickbp** [comment]
retest this please

**Comment by vargup** [comment]
retest this please

**Comment by nickbp** [comment]
retest this please

**Comment by vargup** [comment]
retest this please

**Comment by vargup** [comment]
retest this please

**Comment by vargup** [comment]
can you please look into why builds were failing and this one is running for almost a day?

**Comment by mgummelt** [comment]
@vargup Looks like an intermittent jenkins failure.  retesting...

**Comment by mgummelt** [comment]
retest this please

**Comment by mgummelt** [comment]
retest this please

**Comment by triclambert** [comment]
@mesosphere-ci retest this please

**Comment by vargup** [comment]
retest this please

**Comment by vargup** [comment]
retest this please

**Comment by vargup** [comment]
disk tests have failed twice, can you please check what is going wrong?

**Comment by NimaVaziri** [comment]
The fix for the error has been merged, so rebasing this on master should make the tests pass.

**Comment by vargup** [comment]
@NimaVaziri is the fix merged in the master?

**Comment by NimaVaziri** [comment]
Yeah it is

**Comment by vargup** [comment]
retest this please

**Comment by NimaVaziri** [comment]
looks like master wasn't merged with this branch before retesting

**Comment by verma7** [comment]
Merged PR as all the checks have successfully passed.

**Comment by NimaVaziri** [comment]
👍

**Pull request merged**